### ISSUE TITLE
[Question][C++] State of the old input functions today ?

### ISSUE THREAD

**Issue by ahtur** [ValveSoftware/openvr#1008]
Hi all. I'm totally new to the API (but not to C++ nor to VR). I'm developing an application for Windows 64 bits that reads the **positions and states** of multiple Vive controllers and trackers. I'd be more confortable getting direct values through IVRSystem than through IVRInput and the cumbersome action binding process that it involves.

Could someone from the dev team please tell whether IVRSystem's tracking and controller methods, including those declared as deprecated, still work or not **with the latest SteamVR update** and what are the plans for the future ? I'd really appreciate it if developers had the possibility not to work with actions if they don't want to.

**Comment by bddckr** [comment]
Look into the comments/documentation found in [the C header](https://github.com/ValveSoftware/openvr/blob/master/headers/openvr.h):
>This function is deprecated in favor of the new IVRInput system

**Comment by BOLL7708** [comment]
I know one of my C# applications that loaded controller data through `IVRSystem.PollNextEvent` (through the C# header) stopped working completely for input events sometime around September last year. I can still get events for SteamVR in general, like if a device connected/disconnected etc, but for specifically input like button presses it seems the new input system is mandatory.

I've not had much success implementing it into my background applications though, I have a few issues registered here about that. Background applications seemingly don't behave the way that's necessary to register them for input and edit the bindings in the interface. 😒

**Comment by peroht** [comment]
I expect more functions to be deprecated whenever and if I was you I'd move to the new IVRInput system as soon as possible. There is very little info on when these things are completely deleted. 

I personally think that the flexibility of the new system far outweighs filtering between device numbers here and there. I find it especially annoying when applications use e.g. device number 0/1 and believe they will always be the left/right controller. I have found that this clashes alot with other custom controllers or trackers etc and becomes especially troublesome when we as developers put a requirement on users to start controllers in specific sequences. 

Meta data is the way to go in my opinion.

**Comment by bddckr** [comment]
> There is very little info on when these things are completely deleted.

https://github.com/ValveSoftware/openvr/issues/1090#issuecomment-481300979

**Comment by peroht** [comment]
Thanks for the info @bddckr . It is really good to know.

**Comment by ahtur** [comment]
> I expect more functions to be deprecated whenever and if I was you I'd move to the new IVRInput system as soon as possible. There is very little info on when these things are completely deleted.
> 
> I personally think that the flexibility of the new system far outweighs filtering between device numbers here and there. I find it especially annoying when applications use e.g. device number 0/1 and believe they will always be the left/right controller. I have found that this clashes alot with other custom controllers or trackers etc and becomes especially troublesome when we as developers put a requirement on users to start controllers in specific sequences.
> 
> Meta data is the way to go in my opinion.

Yes, since my question, we've implemented the actions paradigm to handle all buttons. Still wondering what will to happen to the tracking data, though, whether it'll stay in IVISystem or become some sort of "pose action" too...

**Comment by peroht** [comment]
You can access tracking data via the vr::VRInput()->GetPoseActionData function.
